File: Lexor/Lexor.Model/Responses/RFIDResponse.cs
namespace Lexor.Model.Responses
{
    public class RFIDResponse
    {
        public int Id { get; set; }
        public EmployeeResponse Employee { get; set; } = null!;
        public string Uid { get; set; } = string.Empty;
        public DateTime AssignedAt { get; set; }
        public DateTime? DeactivatedAt { get; set; }
        public bool IsActive { get; set; }

        public class EmployeeResponse
        {
            public int Id { get; set; }
            public UserResponse User { get; set; } = null!;
        }

        public class UserResponse
        {
            public int Id { get; set; }
            public string FirstName { get; set; } = string.Empty;
            public string LastName { get; set; } = string.Empty;

            // Small avatar so list views can show the person's photo next to their name
            // (guideline 6) without carrying the full-size image (guideline 8.2).
            public string? ProfileThumbnailBase64 { get; set; }
        }
    }
}

File: Lexor/UI/lexor_desktop/lib/models/rfid_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'rfid_response.g.dart';

@JsonSerializable(createToJson: false)
class RfidResponse {
  final int id;
  final EmployeeResponse employee;
  final String uid;
  final DateTime assignedAt;
  final DateTime? deactivatedAt;
  final bool isActive;

  RfidResponse({
    required this.id,
    required this.employee,
    required this.uid,
    required this.assignedAt,
    required this.isActive,
    this.deactivatedAt,
  });

  String get employeeFullName =>
      '${employee.user.firstName} ${employee.user.lastName}';

  /// Avatar for list views; null when there is no photo.
  String? get employeeThumbnail => employee.user.profileThumbnailBase64;

  factory RfidResponse.fromJson(Map<String, dynamic> json) =>
      _$RfidResponseFromJson(json);
}

@JsonSerializable(createToJson: false)
class EmployeeResponse {
  final int id;
  final UserResponse user;

  EmployeeResponse({required this.id, required this.user});

  factory EmployeeResponse.fromJson(Map<String, dynamic> json) =>
      _$EmployeeResponseFromJson(json);
}

@JsonSerializable(createToJson: false)
class UserResponse {
  final int id;
  final String firstName;
  final String lastName;

  /// Small avatar sent with list responses; null when the person has no photo.
  final String? profileThumbnailBase64;

  UserResponse({
    required this.id,
    required this.firstName,
    required this.lastName,
    this.profileThumbnailBase64,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) =>
      _$UserResponseFromJson(json);
}
